apps/web/server/lib/[user]/getServerSideProps.ts

```diff
@@ -106,7 +106,7 @@ export const getServerSideProps: GetServerSideProps<UserPageProps> = async (cont
   log.debug(safeStringify({ usersInOrgContext, isValidOrgDomain, currentOrgDomain, isDynamicGroup }));
 
   if (isDynamicGroup) {
-    const destinationUrl = `/${usernameList.join("+")}/dynamic`;
+    const destinationUrl = encodeURI(`/${usernameList.join("+")}/dynamic`);
```

Comment:
When you redirect("/franz-janßen"), Next.js redirects you to /franz-jan%DFen
which is using ISO-8859-1 Standards, not UTF-8, and we end up with a throw

Reply:
https://app.cal.com/franz-jan%C3%9Fen+bennyjoo throws:

<img width="1512" alt="Screenshot 2025-02-23 at 1 29 31 AM" src="https://github.com/user-attachments/assets/e245a83a-9e87-4dbe-94b2-fe4b022b64e9" />